### PR TITLE
Add support for getting field names

### PR DESCRIPTION
**Pull request by phd3**
Adds a `fieldNames` method to StdStructType to return list of field names without having to operate on data.

It doesn't seem to be currently used as the names can also be accessible while working with actual data records. e.g. through implementations of `StdStruct#getField` with index or name. 

However, our usecase is to be able to do semantic validations of a JSON-like path representing a field in a dataset schema, using some sort of type representation, and this is one of the things needed for picking transport's type system as that type representation. cc @wmoustafa @khaitranq

### CODE REVIEW

File: transportable-udfs-type-system/src/test/java/com/linkedin/transport/typesystem/AbstractTestStdFactoryForTypes.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+
+public abstract class AbstractTestStdFactoryForTypes {
```
Comment (khaitranq) [inline transportable-udfs-type-system/src/test/java/com/linkedin/transport/typesystem/AbstractTestStdFactoryForTypes.java:20]: Few thoughts:
1. Just wonder if we need to create this new test for testing parsing signature to StdType? It looks like an integration test to me and each individual conversion should be covered in separate unit tests.

If we dont need such integration test, then @phd3 you can just add a few lines to check for the new method you introduced for StdStructType in each type system. For example you can add some test code for that method in [Avro here](https://github.com/linkedin/transport/blob/09a89508296a2491f43cc8866d47952c911313ab/transportable-udfs-avro/src/test/java/com/linkedin/transport/avro/TestAvroWrapper.java) or [Spark here](https://github.com/linkedin/transport/blob/master/transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkFactory.scala). Though I could not find similar tests for Hive and Trino and seems we missed that @wmoustafa?

In this way, @phd3 you dont have to copy the `TrinoTestFunctionDependencies` code here, avoid duplicate code issue?

2. If we still want to do such integration test here, then a minor to rename the test: TestSignatureToStdType for better meaning.

Comment (phd3) [reply transportable-udfs-type-system/src/test/java/com/linkedin/transport/typesystem/AbstractTestStdFactoryForTypes.java:20]: Yeah - seems okay to just add individual unit tests.

**Review by khaitranq — DISMISSED** [review]
LGTM now

**Comment by phd3** [comment]
Thanks for the review @khaitranq .

@wmoustafa @ljfgem would you be able to give this a review?

**Review by ljfgem — APPROVED** [review]
LGTM, thanks @phd3 !
@wmoustafa @aastha25 @yiqiangin do you want to take another look?

File: transportable-udfs-avro/src/test/java/com/linkedin/transport/avro/TestAvroWrapper.java [hunk 2]
```diff
@@ -171,29 +171,32 @@ public void testMapType() {
 
   @Test
   public void testRecordType() {
-    Schema field1 = createSchema("field1", "\"int\"");
-    Schema field2 = createSchema("field2", "\"double\"");
+    Schema field1 = createSchema("fieldOne", "\"int\"");
```
Comment (aastha25) [inline transportable-udfs-avro/src/test/java/com/linkedin/transport/avro/TestAvroWrapper.java:174]: just want to confirm that the change in fieldNames was a cosmetic change and that field names can still be alphanumeric.

**Review by aastha25 — APPROVED** [review]
LGTM

File: transportable-udfs-avro/src/test/java/com/linkedin/transport/avro/TestAvroWrapper.java (hunk 2 above)
Comment (phd3) [reply transportable-udfs-avro/src/test/java/com/linkedin/transport/avro/TestAvroWrapper.java:174]: yes, that's correct. The cosmetic change is for making this a more defensive test w.r.t. case sensitivity.

**Comment by phd3** [comment]
Thanks for the reviews! I've squashed the fixup commit.

**Pull request merged**